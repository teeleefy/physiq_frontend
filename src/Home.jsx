import { Button, Card, CardBody, CardTitle, CardText, ListGroup } from "reactstrap";
import { NavLink } from "react-router-dom";
import { useState, useContext, useEffect } from "react";
import {FamilyContext} from "./auth/UserContext";
import FamilyMember from "./families/FamilyMember";
import PhysiqApi from "./Api.js";
import './styles/Home.css'

function Home(){
    let { currentFamily, setCurrentFamily, setToken } = useContext(FamilyContext);
    const [isLoading, setIsLoading] = useState(true);
      
        useEffect(function loadFamilyInfo() {
          async function getCurrentFamily() {
                try{
                    let family = await PhysiqApi.getCurrentFamily(currentFamily.id);
                    console.log(family);
                    setCurrentFamily(family);
                }catch (err) {
                console.error("App loadUserInfo: problem loading", err);
                // setCurrentMember(null);
                
              }
             setIsLoading(false); 
          }
          getCurrentFamily();
        }, []);

    function renderFamilyMembers(){
        if(currentFamily.familyMembers){
            return (
                <>
                 {currentFamily.familyMembers.map(member => 
                    (<FamilyMember familyMember={member} key={member.id}/>))}
                </>
            )
        }
    }

    function renderHomeMessage(){
        if(currentFamily){
            return(
                <div>
                    <h1 className="Home-title text-center">{currentFamily.name} Dashboard</h1> 
                <ListGroup>
                  {renderFamilyMembers()}  
                </ListGroup>
                <NavLink to={`/add`}>
                    <Button className="Home-addMember btn-dark m-3">Add Member</Button>
                </NavLink>
                </div>
            )
        }
        return (
            <div className="Home-btns">
            <CardTitle className="text-center">
                <h1 className="Home-title">Welcome to fa<b className='mychart'>m</b>il<b className='mychart'>y</b> <b className='mychart'>chart</b>!</h1>
            </CardTitle>
            <CardText className="Home-text">
                Keep track of your family's health, all in one, convenient place!
            </CardText>

            <NavLink to="/login" >
                <Button className="Home-btn">Login</Button>
            </NavLink>
                      
           <NavLink to="/signup" >
                <Button className="Home-btn justify-content-center">Sign Up</Button>
            </NavLink>
            </div>
        )
    }

 return(
    <div id="Home-main">
        <div id="Home-message">

            {renderHomeMessage()} 
        </div>
           
    </div>
 )
}

export default Home;