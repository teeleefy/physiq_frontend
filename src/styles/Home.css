#Home-main {
    width: 90vw;
    height: 82vh;
    background-color: white;
    overflow: scroll;
    color: black;
    border: 15px red solid;
    border-radius: 15px;
    padding: 0;
    margin: 0;
  }

.Home-title{
    margin-bottom: 1rem;
    font-weight: 600;
    text-shadow: 2px 2px 2px grey;
    font-size: 3rem; 
    padding-top: 1rem;
}

.Home-text {
    font-size: 1.5rem; 
}

.Home-btns{
    text-align: center;
}

.Home-btn{
    color: black !important;
    text-decoration: none;
    font-size: 2.5rem !important;
    font-weight: bold !important;
    transition: color 0.5s;
    margin: 1rem; 
    padding: 1rem;
    background-color: red !important;
    box-shadow: 5px 5px 8px black;
    border:0 !important;
    width: 20vw !important;
    transition: all 0.6s !important;
}

.Home-btn:hover{
    background-color: black !important;
    box-shadow: 5px 5px 8px red;
    width: 23vw !important;
    font-size: 2.7rem !important;
    color:red !important;
}

.Home-addMember{
    color: white !important;
    font-size: 1.5rem;
    text-decoration: none;
    width:12vw;
}

.Home-familymember{
    background-color: red !important;
    width: 50vw;
    margin: auto;
    border:0 !important;
}