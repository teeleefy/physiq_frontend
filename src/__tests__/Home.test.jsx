import { renderWithRouterAndFamilyContext, renderWithRouterAndLoggedOutContext, screen } from '../../__tests__/test-tools';
import Home from '../Home'

describe('Home when logged out', () => {
  it('renders the Home component when logged out', () => {
    renderWithRouterAndLoggedOutContext(<Home />)
    
    screen.debug(); // prints out the jsx in the Home component unto the command line
  })

  it("matches snapshot when logged out", function() {
    const { asFragment } = renderWithRouterAndLoggedOutContext(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
})

describe('Home with Family Context', () => {
  it('renders the Home component', () => {
    renderWithRouterAndFamilyContext(<Home />)
    
    screen.debug(); // prints out the jsx in the Home component unto the command line
  })

  it("matches snapshot", function() {
    const { asFragment } = renderWithRouterAndFamilyContext(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
})


