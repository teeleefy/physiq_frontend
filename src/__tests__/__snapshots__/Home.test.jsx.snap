// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="Home-main"
  >
    <div
      id="Home-message"
    >
      <div>
        <h1
          class="Home-title text-center"
        >
          The Smith Family Dashboard
        </h1>
        <ul
          class="list-group"
        />
        <a
          class=""
          href="/add"
        >
          <button
            class="Home-addMember btn-dark m-3 btn btn-secondary"
          >
            Add Member
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home when logged out > matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    id="Home-main"
  >
    <div
      id="Home-message"
    >
      <div
        class="Home-btns"
      >
        <div
          class="text-center card-title"
        >
          <h1
            class="Home-title"
          >
            Welcome to fa
            <b
              class="mychart"
            >
              m
            </b>
            il
            <b
              class="mychart"
            >
              y
            </b>
             
            <b
              class="mychart"
            >
              chart
            </b>
            !
          </h1>
        </div>
        <p
          class="Home-text card-text"
        >
          Keep track of your family's health, all in one, convenient place!
        </p>
        <a
          class=""
          href="/login"
        >
          <button
            class="Home-btn btn btn-secondary"
          >
            Login
          </button>
        </a>
        <a
          class=""
          href="/signup"
        >
          <button
            class="Home-btn justify-content-center btn btn-secondary"
          >
            Sign Up
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home with Family Context > matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="Home-main"
  >
    <div
      id="Home-message"
    >
      <div>
        <h1
          class="Home-title text-center"
        >
          The Smith Family Dashboard
        </h1>
        <ul
          class="list-group"
        />
        <a
          class=""
          href="/add"
        >
          <button
            class="Home-addMember btn-dark m-3 btn btn-secondary"
          >
            Add Member
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
